src/game_actor.cpp

```diff
@@ -86,11 +86,21 @@ Game_Actor::Game_Actor(int actor_id) {
 	SetHp(GetMaxHp());
 	SetSp(GetMaxSp());
 
-	SetEquipment(1, dbActor->initial_equipment.weapon_id);
-	SetEquipment(2, dbActor->initial_equipment.shield_id);
-	SetEquipment(3, dbActor->initial_equipment.armor_id);
-	SetEquipment(4, dbActor->initial_equipment.helmet_id);
-	SetEquipment(5, dbActor->initial_equipment.accessory_id);
+	std::array<int, 5> ids = {{
+		dbActor->initial_equipment.weapon_id & 0xFFFF,
+		dbActor->initial_equipment.shield_id & 0xFFFF,
+		dbActor->initial_equipment.armor_id & 0xFFFF,
+		dbActor->initial_equipment.helmet_id & 0xFFFF,
+		dbActor->initial_equipment.accessory_id & 0xFFFF }};
+	std::replace_if(ids.begin(), ids.end(), [] (const int& item_id) {
+		return lcf::ReaderUtil::GetElement(lcf::Data::items, item_id) == nullptr;
+	}, 0);
```

Comment:
This & 0xFFFF is not needed here. This was just because of the garbage data in battle test.

Also add a comment like "// Remove items that do not exist in the database anymore"

Reply:
Fixed.